op-node/rollup/derive/engine_update.go

```diff
@@ -160,6 +173,10 @@ func confirmPayload(ctx context.Context, log log.Logger, eng ExecEngine, fc eth.
 	if fcRes.PayloadStatus.Status != eth.ExecutionValid {
 		return nil, BlockInsertPayloadErr, eth.ForkchoiceUpdateErr(fcRes.PayloadStatus)
 	}
+	// if the block was inserted, clear it from the async gossiper
+	if agossip != nil {
+		agossip.Clear()
+	}
```

Comment:
I suspect we should also clear the gossip block if op-geth reported that it was invalid.  That _shouldn't_ happen but it certainly isn't unheard of for an execution client to build an invalid block for some reason and its important that we don't get stuck repeatedly trying to import it.  If it was already gossiped, the recipients should also have rejected it for being invalid anyway.

The key thing is that we don't want to clear if there was a temporary error like op-geth being offline or timing out. Only if it explicitly says the block is invalid (or if it imported it fine).

Reply:
That sounds fine, and I think it just means that this block moves *above* the `!= eth.ExecutionValid` check. At that point, the payload is either invalid, or inserted, and in either case we are ready to clear. Moved it there, but LMK if I misunderstand.